include/pybind11/pytypes.h

```diff
@@ -214,6 +214,7 @@ class handle : public detail::object_api<handle> {
     /// Creates a ``handle`` from the given raw Python object pointer
     // NOLINTNEXTLINE(google-explicit-constructor)
     handle(PyObject *ptr) : m_ptr(ptr) {} // Allow implicit conversion from PyObject*
```

Comment:
Maybe this is a job for `enable_if`?
I'd bet though `enable_if` `PyObject*` won't work, `is_pointer` or similar might. Or maybe to be fancy, look for `ob_refcnt` in the pointee?

Reply:
@rwgk Hmm... can't figure out the way to construct the enable_if_t.

Reply:
This strategy doesn't work because it causes ambiguous overloads, which is ridiculous since the func is deleted.